src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/WinformsUpdateStep.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Windows
+{
+    /// <summary>
+    /// Upgrade step that updates Winforms Projects.
+    /// </summary>
+    public class WinformsUpdateStep : UpgradeStep
+    {
+        public override string Title => "Update Winforms Project";
+
+        public override string Description => "Update Winforms Project";
+
+        public override string Id => WellKnownStepIds.WinformsProjectUpdaterStepId;
+
+        public override IEnumerable<string> DependsOn { get; } = new[]
+        {
+            // Project should be backed up before changing source
+            WellKnownStepIds.BackupStepId,
+
+            // Template files should be added prior to changing source (since some code fixers will change added templates)
+            WellKnownStepIds.TemplateInserterStepId,
+
+            // Project should have correct TFM
+            WellKnownStepIds.SetTFMStepId,
+        };
+
+        public override IEnumerable<string> DependencyOf { get; } = new[]
+        {
+            WellKnownStepIds.NextProjectStepId,
+        };
+
+        public WinformsUpdateStep(IEnumerable<IUpdater<IProject>> winformsUpdaters, ILogger<WinformsUpdateStep> logger)
+            : base(logger)
+        {
+            if (logger is null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            SubSteps = new List<UpgradeStep>(winformsUpdaters.Select(updater => new WinformsUpdaterSubStep(this, updater, logger)));
```

Comment:
suggestion: Instead of `newing` it up, you're already using LINQ, so just tack on a `.ToList()` call